File: ios/App/App/FreeDiskPlugin.swift
import Foundation
import Capacitor

/* Reports the device's REAL available disk capacity (the value iOS would
   allow an app to use for important data), which WKWebView cannot see.
   Used by the recorder to warn/auto-stop before storage exhausts. */
@objc(FreeDiskPlugin)
public class FreeDiskPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "FreeDiskPlugin"
    public let jsName = "FreeDisk"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "free", returnType: CAPPluginReturnPromise)
    ]

    @objc func free(_ call: CAPPluginCall) {
        let url = URL(fileURLWithPath: NSHomeDirectory())
        do {
            let values = try url.resourceValues(forKeys: [
                .volumeAvailableCapacityForImportantUsageKey,
                .volumeTotalCapacityKey
            ])
            call.resolve([
                "free": values.volumeAvailableCapacityForImportantUsage ?? -1,
                "total": values.volumeTotalCapacity ?? -1
            ])
        } catch {
            call.reject("Free-disk query failed: \(error.localizedDescription)")
        }
    }
}

File: android/app/src/main/java/com/meetingghost/app/FreeDiskPlugin.java
package com.meetingghost.app;

import android.os.StatFs;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/** Reports real app-volume free space for recorder warning and safe auto-stop. */
@CapacitorPlugin(name = "FreeDisk")
public class FreeDiskPlugin extends Plugin {
    @PluginMethod
    public void free(PluginCall call) {
        try {
            StatFs stats = new StatFs(getContext().getFilesDir().getAbsolutePath());
            JSObject result = new JSObject();
            result.put("free", stats.getAvailableBytes());
            result.put("total", stats.getTotalBytes());
            call.resolve(result);
        } catch (Exception error) {
            call.reject("Free-disk query failed", error);
        }
    }
}
